internal/xds/balancer/clusterimpl/tests/balancer_test.go

```diff
@@ -436,21 +440,255 @@ func (s) TestCircuitBreaking(t *testing.T) {
 
 	// Since we are at the max, new streams should fail.  It's possible some are
 	// allowed due to inherent raciness in the tracking, however.
-	for i := 0; i < 100; i++ {
+	const droppedRPCCount = 100
+	for i := 0; i < droppedRPCCount; i++ {
+		_, err := client.FullDuplexCall(ctx)
+		switch {
+		case err == nil:
+			t.Fatalf("client.FullDuplexCall(_) failed with unexpected error got: <nil>, want: %v", err)
+		case status.Code(err) == codes.Unavailable:
+			continue
+		default:
+			t.Errorf("client.FullDuplexCall(_) failed with unexpected error = %v", err)
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
+
+	if _, err = mgmtServer.LRSServer.LRSStreamOpenChan.Receive(ctx); err != nil {
+		t.Fatalf("Failure when waiting for an LRS stream to be opened: %v", err)
+	}
+
+	if _, err = mgmtServer.LRSServer.LRSRequestChan.Receive(ctx); err != nil {
+		t.Fatalf("Failure waiting for initial LRS request: %v", err)
+	}
+
+	resp := fakeserver.Response{
+		Resp: &v3lrspb.LoadStatsResponse{
+			SendAllClusters:       true,
+			LoadReportingInterval: durationpb.New(10 * time.Millisecond),
+		},
+	}
+	mgmtServer.LRSServer.LRSResponseChan <- &resp
+
+	select {
+	case req := <-mgmtServer.LRSServer.LRSRequestChan.C:
+		loadStats := req.(*fakeserver.Request).Req.(*v3lrspb.LoadStatsRequest)
+		for _, cs := range loadStats.ClusterStats {
+			if cs.TotalDroppedRequests != droppedRPCCount {
+				t.Errorf("Failed with unexpected diff got: %v, want: %v", cs.TotalDroppedRequests, droppedRPCCount)
+			}
+			if cs.UpstreamLocalityStats[0].TotalIssuedRequests != maxRequests {
+				t.Errorf("Failed with unexpected diff got: %v, want: %v", cs.UpstreamLocalityStats[0].TotalIssuedRequests, maxRequests)
+			}
+		}
+	case <-ctx.Done():
+		t.Fatalf("Timeout while waiting for LRS stream: %v", ctx.Err())
+	}
+}
+
+// ComputeIdealNumRpcs calculates the ideal number of RPCs to send to test
+// probabilistic behaviors.
+//
+// It's based on the idea of a binomial distribution approximated by a normal
+// distribution. The function aims to find a number of RPCs such that
+// the observed probability is within a certain error_tolerance of the expected
+// probability (p).
+func computeIdealNumRpcs(p float64, errorTolerance float64) uint64 {
+	numRpcs :=
+		uint64(math.Ceil(p * (1 - p) * 5.00 * 5.00 / errorTolerance / errorTolerance))
+	return numRpcs
+}
+
+// TestDropByCategory verifies that the balancer correctly drops the picks, and
+// that the drops are reported.
+func TestDropByCategory(t *testing.T) {
```

Comment:
Same here. Why is this test missing the `s` receiver. The `s` struct embeds a `grpctest.Tester` which has a setup and teardown method that gets run for every test. Please see https://github.com/grpc/grpc-go/blob/master/internal/grpctest/grpctest.go#L52